daemon/attach.go

```diff
@@ -113,132 +113,86 @@ func (daemon *Daemon) ContainerAttach(job *engine.Job) engine.Status {
 
 func (daemon *Daemon) attach(streamConfig *StreamConfig, openStdin, stdinOnce, tty bool, stdin io.ReadCloser, stdout io.Writer, stderr io.Writer) chan error {
 	var (
-		cStdout, cStderr io.ReadCloser
-		nJobs            int
-		errors           = make(chan error, 3)
+		cStdout = streamConfig.StdoutPipe()
```

Comment:
Now, I think why we need this pipes if `stdout` or `stderr` is nil? This is actually pretty heavy pipes, which have a lot of goroutines.